hydrolib/core/io/bc/models.py

```diff
@@ -136,50 +50,107 @@ def _supports_comments(cls):
+
+
+class Harmonic(ForcingBase):
+    function: Literal["harmonic"] = "harmonic"
+    factor: float = 1.0
+
+
+class Astronomic(ForcingBase):
+    function: Literal["astronomic"] = "astronomic"
+    factor: float = 1.0
+
+
+class HarmonicCorrection(ForcingBase):
+    function: Literal["harmoniccorrection"] = "harmoniccorrection"
 
-    function: Union[
-        Function.TimeSeries,
-        Function.Harmonic,
-        Function.Astronomic,
-        Function.HarmonicCorrection,
-        Function.AstronomicCorrection,
-        Function.T3D,
-        Function.QHTable,
-        Function.Constant,
-    ]
+
+class AstronomicCorrection(ForcingBase):
+    function: Literal["astronomiccorrection"] = "astronomiccorrection"
+
+
+class T3D(ForcingBase):
+    function: Literal["t3d"] = "t3d"
+
+    offset: float = 0.0
+    factor: float = 1.0
+
+    vertical_positions: List[float] = Field(alias="verticalPositions")
+    vertical_interpolation: VerticalInterpolation = Field(alias="verticalInterpolation")
+    vertical_position_type: VerticalPositionType = Field(alias="verticalPositionType")
+
+
+class QHTable(ForcingBase):
+    function: Literal["qhtable"] = "qhtable"
+
+
+class Constant(ForcingBase):
+    function: Literal["constant"] = "constant"
+
+    offset: float = 0.0
+    factor: float = 1.0
+
+
+class ForcingGeneral(INIGeneral):
+    fileVersion: str = "1.01"
+    fileType: Literal["boundConds"] = "boundConds"
+
+
+class ForcingModel(INIModel):
+    general: ForcingGeneral
+    forcing: List[ForcingBase]
 
     @classmethod
-    def _convert_section_content(cls, content: List) -> Dict:
-        content_dict = super()._convert_section_content(content)
-        cls._group_function_data(content_dict)
-        return content_dict
+    def _ext(cls) -> str:
+        return ".bc"
 
     @classmethod
-    def _convert_section(cls, section: Section) -> Dict:
-        result = super()._convert_section(section)
-        transfer_keyvalue(result, result["function"], "datablock")
-        return result
+    def _filename(cls) -> str:
+        return "forcing"
 
     @classmethod
-    def _group_function_data(cls, data: Dict) -> None:
-        function_data = {"function": data["function"]}
-
-        keys_to_transfer = [
-            "offset",
-            "factor",
-            "verticalPositions",
-            "verticalInterpolation",
-            "verticalPositionType",
-            "timeInterpolation",
-            "quantity",
-            "unit",
-            "verticalPositionIndex",
-        ]
-
-        for key in keys_to_transfer:
-            transfer_keyvalue(data, function_data, key)
-
-        data["function"] = function_data
+    def _get_parser(cls) -> Callable:
+        return cls.parse
+
+    @classmethod
+    def parse(cls, filepath: Path):
+        # It's odd to have to disable parsing something as comments
+        # but also need to pass it to the *flattener*.
+        # This method now only supports per model settings, not per section.
```

Comment:
This is by design, such settings are defined per whole file, not per section.

Reply:
This is the documentation of the design, should be followed up by #40